fastfood/food.py

```diff
@@ -92,15 +87,26 @@ def update_cookbook(cookbook, templatepack, stencilset_name, **options):
     files = {
         # files.keys() are template paths, files.values() are target paths
         # {path to template: rendered target path, ... }
-        os.path.join(stencilset.path, tpl): os.path.join(cookbook.path, tgt)
+        os.path.join(stencilset.path, tpl): os.path.join(cookbook_path, tgt)
         for tgt, tpl in stencil['files'].iteritems()
     }
 
     template_map = {
+        'cookbook': {"name": cookbook_name},
         'options': stencil['options'],
-        'cookbook': cookbook.metadata.to_dict().copy(),
     }
 
+    # Cookbooks may not yet have metadata, so we pass an empty dict if so
+    try:
+        cookbook = book.CookBook(cookbook_path)
+        template_map['cookbook'] = cookbook.metadata.to_dict().copy()
+    except ValueError as err:
```

Comment:
Since only the `cookbook.metadata` piece is "allowed" to raise the `ValueError`, lets move the line `cookbook = book.CookBook(cookbook_path)` to above the `try` block, up to line ~99